src/Switch.tsx

```diff
@@ -0,0 +1,358 @@
+  }};
+  border-radius: 6px;
+  display: block;
+  height: 32px;
+  width: 64px;
+  position: relative;
+  outline-offset: 1px;
+
+  @media (pointer: coarse) {
+    &:before {
+      content: '';
+      position: absolute;
+      left: 0;
+      right: 0;
+      transform: translateY(-50%);
+      top: 50%;
+      min-height: 44px;
+    }
+  }
+
+  &:after {
+    content: '';
+    box-sizing: border-box;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    border-radius: 6px;
+  }
+
+  ${props => {
+    if (!props.isHighContrast) {
+      return css`
+        &:after {
+          border-style: solid;
+          border-width: 1px;
+          border-color: ${props.on && !props.disabled ? get('colors.accent.fg') : get('colors.border.subtle')};
+          opacity: ${props.disabled ? '0.5' : props.on ? '0.2' : '1'};
+        }
+      `
+    } else {
+      return css`
+        border-style: solid;
+        border-color: ${props.on && !props.disabled ? get('colors.accent.fg') : get('colors.border.subtle')};
+        border-width: 1px;
+        outline-offset: 2px;
+
+        .Toggle-knob {
+          left: -1px;
+          top: -1px;
+          bottom: -1px;
+          transform: translateX(${props.on ? 'calc(100% + 1px)' : '0'});
+        }
+      `
+    }
+  }}
+
+  ${props => {
+    if (!props.disabled) {
+      if (props.on) {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.2;
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            :after {
+              background-color: ${get('colors.accent.muted')};
+              opacity: 0.5;
+            }
+          }
+        `
+      } else {
+        return css`
+          &:hover,
+          &:focus:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.hoverBg')};
+              border-color: ${get('colors.border.default')};
+            }
+          }
+
+          &:active,
+          &:active:focus-visible {
+            .Toggle-knob {
+              background-color: ${get('colors.btn.activeBg')};
+            }
+          }
+        `
+      }
+    }
+  }}
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+
+  ${sx}
+  ${sizeVariants}
+`
+
+const hiddenTextStyles: BetterSystemStyleObject = {
+  visibility: 'hidden',
+  height: 0
+}
+
+const Switch: React.FC<SwitchProps> = ({
+  'aria-labelledby': ariaLabelledby,
+  'aria-describedby': ariaDescribedby,
+  defaultOn,
+  disabled,
+  isLoading,
+  on: onProp,
+  onChange,
+  onClick,
+  size,
+  statusLabelPosition,
+  sx: sxProp
+}) => {
+  const isControlled = typeof defaultOn === 'undefined' && typeof onProp !== 'undefined'
+  const {theme, colorScheme} = useTheme()
+  const [onState, setOnState] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
```

Comment:
`isOn` is a little clearer to me. `onState` makes me think it's an event handler

```suggestion
  const [isOn, setIsOn] = useProvidedStateOrCreate<boolean>(onProp, onChange, Boolean(defaultOn))
```